lib/ruby_lsp/requests/formatting.rb

```diff
@@ -67,16 +92,10 @@ def run
 
       sig { returns(T.nilable(String)) }
       def formatted_file
-        case @formatter
-        when "rubocop"
-          if defined?(Support::RuboCopFormattingRunner)
-            Support::RuboCopFormattingRunner.instance.run(@uri, @document)
-          end
-        when "syntax_tree"
-          Support::SyntaxTreeFormattingRunner.instance.run(@uri, @document)
-        else
-          raise InvalidFormatter, "Unknown formatter: #{@formatter}"
-        end
+        formatter = Formatting.formatters[@formatter]
```

Comment:
(nit) I have a slight aversion to having a local variable with the same name as an instance variable... maybe this could be `formatter_instance`?